### PR TITLE
More informative error for BlockNotFound exception for debugging

### PR DESCRIPTION
**Pull request by hbdeshmukh**

### CODE REVIEW

**Comment by jianqiao** [comment]
The change looks good! Some minor fixes are needed.

File: storage/StorageErrors.hpp [hunk 1]
```diff
@@ -61,9 +61,16 @@ class BlockMemoryTooSmall : public std::exception {
  **/
 class BlockNotFoundInMemory : public std::exception {
  public:
+  BlockNotFoundInMemory(int block_id) : block_id_(block_id) {}
+
   virtual const char* what() const throw() {
-    return "BlockNotFoundInMemory: The specified block was not found in memory";
+    std::string message = "BlockNotFoundInMemory: The specified block with ID "
+      + std::to_string(block_id_ )+ " was not found in memory";
+    return message.c_str();
```
[the remark this answers is not in the document]
Comment (hbdeshmukh) [reply storage/StorageErrors.hpp]: My bad, that's a stupid mistake. Thanks for pointing out.

**Comment by hbdeshmukh** [comment]
Hi @jianqiao : Made the changes as requested.

**Comment by jianqiao** [comment]
LGTM! Merging.

**Pull request merged**